app/controllers/api/v1/visualization/clusters_controller.rb

```diff
@@ -6,6 +6,7 @@ module Visualization
       class ClustersController < ApiBaseController
         include Concerns::Authenticator
         include Concerns::StudyAware
+        include Concerns::ApiCaching
```

Comment:
Do we have to worry about the order of this include, so that it's before/after the check_study_view_permission? since it renders out the response on a cache hit, I think this needs to be included after we call before_action :check_study_view_permission so that the auth check happens first

Reply:
That's a really great point - let me do some reading/testing to see what order concern methods fire in based off of includes.

Reply:
As I suspected, you're correct - the order of declaration sets order of precedence for before/around/after filters:

https://stackoverflow.com/questions/34609237/order-of-before-filters-in-rails-controller-and-concerns

Also, I put in some logging statements in each concern, and it backed up that assertion:

```
Started GET "/single_cell/api/v1/studies/SCP16/clusters/Cluster%201?annotation_name=Cluster&annotation_scope=study&annotation_type=group" for 172.17.0.1 at 2020-11-25 10:18:53 -0500
Processing by Api::V1::Visualization::ClustersController#show as JSON
  Parameters: {"annotation_name"=>"Cluster", "annotation_scope"=>"study", "annotation_type"=>"group", "study_id"=>"SCP16", "cluster_name"=>"Cluster 1", "cluster"=>{}}
calling check_api_cache!
calling set_current_api_user!
calling current_api_user
calling set_study
calling check_study_view_permission
calling write_api_cache!
Writing to API cache: _single_cell_api_v1_studies_SCP16_clusters_Cluster_1_annotation_name_Cluster_annotation_scope_study_annotation_type_group_cluster_name_Cluster_1
Completed 200 OK in 1054ms (Views: 1.1ms)
```

Based off of this, I think the "correct" thing to do is to modify the concern to not automatically run the before/after filters on all included modules.   This way, we can control what/when methods fire in each controller like we do for `StudyAware` and `Authenticator`.

Reply:
46b4817